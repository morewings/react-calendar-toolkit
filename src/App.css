.app {
  width: 500px;
  margin: 50px auto;
  display: flex;
  justify-content: space-between;
}

.app .container {
  padding-top: 50px;
}
