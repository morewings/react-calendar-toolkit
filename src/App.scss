.app{
  width: 500px;
  margin: 50px auto;
}