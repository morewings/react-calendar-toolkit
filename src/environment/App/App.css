.app {
  display: flex;
  justify-content: space-between;
  margin: 50px auto;
  width: 500px;
}

.app .container {
  padding-top: 50px;
}
