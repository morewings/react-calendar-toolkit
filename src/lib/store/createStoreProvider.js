import React, {useReducer} from 'react';

const createStoreProvider = ({initialState, reducer, context}) => ({
  // eslint-disable-next-line react/prop-types
  children,
}) => {
  const {Provider} = context;
  const [state, dispatch] = useReducer(reducer, initialState);
  return <Provider value={{state, dispatch}}>{children}</Provider>;
};

export default createStoreProvider;
