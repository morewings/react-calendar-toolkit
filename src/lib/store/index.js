export {default} from 'lib/store/createStoreProvider';
