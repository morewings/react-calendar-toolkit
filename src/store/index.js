// export {default, rootReducer} from './store';
export {default as withDatepickerProvider} from './withDatepickerProvider';
export {default as withModalProvider} from './withModalProvider';
