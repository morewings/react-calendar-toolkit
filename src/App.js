/*eslint-disable*/
import React from 'react';
import {isSameDay} from 'date-fns'
import DatePicker from 'components/DatePicker';
import DatePickerFieldset from 'components/DatePickerFieldset';
import './App.scss';

const highlightDate = ({date, precision}) => {
  return isSameDay(date, new Date(2020, 0, 7))
};

const App = () => (
  <div className="app">
    <div className="container">
      <DatePicker
        startDate={new Date(1999, 0, 1)}
        endDate={new Date(2020, 1, 25)}
        minPrecision="year"
        highlightDate={highlightDate}
        onDateSet={date => {
          console.log('date set', date);
        }}
        title="Demo datepicker"
      />
    </div>
    <div className="container">
      <DatePickerFieldset
        onDateSet={date => {
          console.log('date set', date);
        }}
        minPrecision="day"
        title="Demo datepicker"
      />
    </div>
  </div>
);

export default App;
