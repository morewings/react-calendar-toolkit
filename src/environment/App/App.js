import React from 'react';
import {isSameDay} from 'date-fns';
import customLocale from 'date-fns/locale/es';
import {Datepicker, DatepickerInput} from 'lib';
import './App.css';

const highlightDate = ({date, precision}) =>
  isSameDay(date, new Date(2020, 0, 7));

const App = () => (
  <div className="app">
    <div className="container">
      <Datepicker
        startDate={new Date(1999, 11, 6)}
        endDate={new Date(2020, 1, 25)}
        minPrecision="hour"
        highlightDate={highlightDate}
        onDateSet={date => {
          // eslint-disable-next-line no-console
          console.log('date set', date);
        }}
        title="Demo datepicker"
        theme={{
          headerBgColor: 'purple',
        }}
      />
    </div>
    <div className="container">
      <DatepickerInput
        mode="popover"
        minPrecision="month"
        onDateSet={date => {
          // console.log('date set', date);
        }}
        dateFnsLocale={customLocale}
        title="Demo datepicker"
      />
    </div>
  </div>
);

export default App;
