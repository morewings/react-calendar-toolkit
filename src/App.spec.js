import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import store from './store';
import App from './App';

it.skip('renders without crashing', () => {
  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
