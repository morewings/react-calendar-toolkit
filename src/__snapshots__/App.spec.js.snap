// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <div
      className="app"
    >
      <Calendar>
        <div>
          Calendar
        </div>
      </Calendar>
    </div>
  </App>
</Provider>
`;
